### PR TITLE
auth-backend: migrated AWS ALB provider to new backend.

### PR DESCRIPTION
**Pull request by GuptaNavdeep1983**
## Hey, I just made a Pull Request!

This migrates the AWS ALB auth provider module.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by GuptaNavdeep1983** [comment]
> Regarding yarn.lock, a hint: you may want to squash your PR, and then `git checkout master yarn.lock` to get exactly the original lock file and then run `yarn install`. There's usually little value in merging it manually.

@freben, following your hint, squashed the PR but its not allowing me to make a commit with the error that its an empty commit even though there is a small change in "@octokit/openapi-types@npm:^17.2.0" package.

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     ]
   },
   "resolutions": {
+    "csstype": ">=3.0.2 <=3.1.2",
```
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply package.json]: updated.

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 🎉 

Just a few small tweaks

**Review by Rugvip — CHANGES_REQUESTED** [review]
derp 😅

File: plugins/auth-backend-module-aws-alb-provider/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+export type AwsAlbResult = {
+  fullProfile: PassportProfile;
+  expiresInSeconds?: number;
+  accessToken: string;
+};
+/**
+ * @public
+ */
+export type PassportProfile = passport.Profile & {
```
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply plugins/auth-backend-module-aws-alb-provider/src/types.ts]: Moved.

File: plugins/auth-backend-module-aws-alb-provider/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "@backstage/backend-common": "workspace:^",
+    "@backstage/backend-plugin-api": "workspace:^",
+    "@backstage/errors": "workspace:^",
+    "@backstage/plugin-auth-backend": "workspace:^",
+    "@backstage/plugin-auth-node": "workspace:^",
+    "@types/passport": "^1.0.3",
+    "jose": "^4.6.0",
+    "jwt-decode": "^3.1.0",
+    "node-cache": "^5.1.2",
+    "passport": "^0.7.0"
```
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply plugins/auth-backend-module-aws-alb-provider/package.json]: Removed.

File: plugins/auth-backend-module-aws-alb-provider/src/index.d.ts
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply plugins/auth-backend-module-aws-alb-provider/src/index.d.ts:1]: Removed.

File: yarn.lock
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply yarn.lock:1]: Performed this step multiple times but it always yield so many changes!

**Comment by benjdlambert** [comment]
@GuptaNavdeep1983 I think this needs another rebase before it will start to build. Looks like there's some conflicts in the yarn.lock.

**Comment by GuptaNavdeep1983** [comment]
> @GuptaNavdeep1983 I think this needs another rebase before it will start to build. Looks like there's some conflicts in the yarn.lock.

No luck still. Same result.

**Review by Rugvip — APPROVED** [review]
Alright did the `yarn.lock` update. This was on top of a pretty old master which was causing the issues. Had to grab it from the merge base instead:

```bash
git checkout $(git merge-base HEAD master) -- yarn.lock
yarn install                                                                                                          
```

**Pull request merged**